build.gradle

```diff
@@ -243,7 +243,7 @@ String bwcFilePath = "src/test/resources/org/opensearch/knn/bwc/"
             }))
             setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
             setting 'http.content_type.required', 'true'
-            systemProperty "java.library.path", "$rootDir/src/test/resources/org/opensearch/knn/bwc/lib:$rootDir/jni/release"
+            systemProperty "java.library.path", "$rootDir/src/test/resources/org/opensearch/knn/bwc/lib:$rootDir/jni/release:$rootDir/$bwcFilePath/$opensearch_bwc_version-SNAPSHOT/opensearch-knn-$opensearch_bwc_version-linux-x64/knnlib"
```

Comment:
We don't need to add this path here. We can copy the dependencies of knnlib(libopensearchknn_common.so, libopensearchknn_faiss.so, libopensearchknn_nmslib.so) into this path "src/test/resources/org/opensearch/knn/bwc/lib" 

Also, I have a doubt here. Can it unzip this file opensearch-knn-1.2.0.0-linux-x64.zip and look for knnlib if we provide the path like this "opensearch-knn-$opensearch_bwc_version-linux-x64/knnlib"?

Reply:
> We don't need to add this path here. We can copy the dependencies of knnlib(libopensearchknn_common.so, libopensearchknn_faiss.so, libopensearchknn_nmslib.so) into this path "src/test/resources/org/opensearch/knn/bwc/lib"

I prefer just unzipping and appending to the path as opposed to unzipping and copying the contents of knnlib into lib. What advantage does copying have?

> Also, I have a doubt here. Can it unzip this file opensearch-knn-1.2.0.0-linux-x64.zip and look for knnlib if we provide the path like this "opensearch-knn-$opensearch_bwc_version-linux-x64/knnlib"?

Let me check on this. The CI passes, but that may be because it is reading the lib from the jni/release folder - this would be problematic because it would always pick up the latest version of the library (as opposed to the bwc version)

Reply:
> 

We already have "src/.../bwc/lib" folder where we can add all the knnlib dependencies of all BWC versions. It will load everything and use the one which is required. If we keep adding paths for all the versions, the code may not look good.

Reply:
>  If we keep adding paths for all the versions

This is parametrized? There are not hardcoded versions in the path.

Reply:
The challenge is to have the old node use 1.2 libs and the new node use the libs in jni/release during mixed cluster.

Reply:
>Sorry, my bad

Reply:
> The challenge is to have the old node use 1.2 libs and the new node use the libs in jni/release during mixed cluster.

Will there be any change in the knnlibs for 1.2 and 1.3?

Reply:
There might be - maybe bug fixes, or optimizations. Nodes running current version should set path to `jni/release` while nodes running a bwc version should point to knnlib in bwc. Any ideas how this could be done @naveentatikonda?